baremetal/metal3machine_manager.go

```diff
@@ -392,14 +392,11 @@ func (m *MachineManager) Associate(ctx context.Context) error {
 	return nil
 }
 
-// getUserData gets the UserData from the machine and exposes it as a secret
-// for the BareMetalHost. The UserData might already be in a secret with
+// getUserDataSecretName gets the UserDataSecretName from the machine and exposes it as a secret
+// for the BareMetalHost. The UserDataSecretName might already be in a secret with
```

Comment:
It is unconditionally related and IMO we have to take these changes in with this PR. To give some more insight on it, we are accessing [Data](https://github.com/metal3-io/cluster-api-provider-metal3/blob/master/baremetal/metal3machine_manager.go#L636) field in CAPI machine (m.Machine.Spec.Bootstrap.Data) as follows:

                                                `if m.Machine.Spec.Bootstrap.DataSecretName == nil &&
			                                    m.Machine.Spec.Bootstrap.Data != nil {`
                                                
and in CAPI v1a4 [Bootstrap.Data](https://pkg.go.dev/sigs.k8s.io/cluster-api@v0.4.0-beta.0/api/v1alpha4#Bootstrap) field was deprecated and suggested to switch to **Bootstrap.DataSecretName**, hence it will complain as it is obviously with:

`m.Machine.Spec.Bootstrap.Data undefined (type "sigs.k8s.io/cluster-api/api/v1alpha4".Bootstrap has no field or method Data)compilerMissingFieldOrMethod`. For that reason I had to remove and introduce those changes and remove according test case for **Bootstrap.Data**. Is there any other way to go around this without these changes?

Reply:
OK. Let's take it in then